services/QuillLMS/client/app/bundles/Teacher/components/classrooms/edit_student_account_modal.tsx

```diff
@@ -11,115 +11,118 @@ interface EditStudentAccountModalProps {
+        setErrors(body.errors)
+        setTimesSubmitted(timesSubmitted + 1)
       } else {
         onSuccess('Student account saved')
         close()
       }
     })
   }
 
-  submitButtonClass() {
-    const { student } = this.props
-    const { firstName, lastName, username } = this.state
-    let buttonClass = 'quill-button-archived contained primary medium';
+  function submitButtonClass() {
+    let buttonClass = 'quill-button contained green medium focus-on-light';
     const originalSplitName = student.name.split(' ')
-    const nothingHasChanged = originalSplitName[0] === firstName && originalSplitName[1] === lastName && student.username === username
-    if (!firstName.length || !lastName.length || !username.length || nothingHasChanged) {
+    const noChanges = originalSplitName[0] === firstName && originalSplitName[1] === lastName && student.username === username
+    if (!firstName.length || !lastName.length || !username.length || noChanges) {
       buttonClass += ' disabled';
     }
     return buttonClass;
   }
 
-  render() {
-    const { firstName, lastName, username, errors, timesSubmitted } = this.state
-    return (
-      <div className="modal-container edit-student-account-modal-container">
-        <div className="modal-background" />
-        <div className="edit-student-account-modal quill-modal modal-body">
-          <div>
-            <h3 className="title">Edit your student account</h3>
-          </div>
-          <Input
-            characterLimit={50}
-            className="firstName"
-            error={errors.firstName}
-            handleChange={this.handleFirstNameChange}
-            label="First name"
-            timesSubmitted={timesSubmitted}
-            type="text"
-            value={firstName}
-          />
-          <Input
-            characterLimit={50}
-            className="lastName"
-            error={errors.lastName}
-            handleChange={this.handleLastNameChange}
-            label="Last name"
-            timesSubmitted={timesSubmitted}
-            type="text"
-            value={lastName}
-          />
-          <Input
-            className="username"
-            error={errors.username}
-            handleChange={this.handleUsernameChange}
-            label="Username"
-            timesSubmitted={timesSubmitted}
-            type="text"
-            value={username}
-          />
-          <div className="form-buttons">
-            <button className="quill-button-archived outlined secondary medium" onClick={this.props.close}>Cancel</button>
-            <button className={this.submitButtonClass()} onClick={this.editStudentAccount}>Save</button>
-          </div>
+  function handleInputChange(e) {
+    const { target } = e
+    const { id, value } = target
+    if (id === FIRST_NAME) {
+      setFirstName(value)
+    }
+    if (id === LAST_NAME) {
+      setLastName(value)
+    }
+    if (id === USERNAME) {
+      setUsername(value)
+    }
```

Comment:
This is purely stylistic, so take or leave, but I prefer objects with variable keys for these sorts of patterns.  It mostly just reads better to me than a sequence of `if` blocks.  I think that's because it makes it clear up-front that all of these are mutually exclusive, I don't have to go through each condition to understand that:
```javascript
const functionsById = {
  [FIRST_NAME]: setFirstName,
  [LAST_NAME]: setLastName,
  [USERNAME]: setUsername
}
functionsById[id](value)
```